Src/Plugins/SalModernization/SalModernization.SharedStrings.txt

```diff
@@ -21,636 +21,632 @@
+$SEC105/546.PreDerefINTMINAccessWrite=\[Pre\( Deref=INT_MIN, Access=Write \)\]
+$SEC105/547.PreDeref1ValidSAYes=\[Pre\(Deref=-1, Valid=SA_Yes\)\]
+$SEC105/548.PreDerefderefAccess1=\[Pre\( Deref=deref, Access=%1 \)\]
+$SEC105/549.PreDerefderefNull1=\[Pre\( Deref=deref, Null=%1 \)\]
+$SEC105/550.PreDerefderefValidvalidValuePreDerefderefderefNNullnullValue=\[Pre\( Deref=deref, Valid=validValue \), Pre\( Deref=deref+derefN, Null=nullValue \)\]
+$SEC105/551.PreNeedsRelease1=\[Pre\( NeedsRelease=%1 \)\]
+$SEC105/552.PreValidSANoAccess=\[Pre\(Valid=SA_NoAccess\)\]
+$SEC105/553.Post=\[Post\(\)\]
+$SEC105/554.Post12=\[Post\(1,2\)\]
+$SEC105/555.Post1NullYes=\[Post\(1, Null=Yes\)\]
+$SEC105/556.PostAccessRead=\[Post\( Access=Read \)\]
+$SEC105/557.PostAccessSAWriteSANoAccess=\[Post\(Access=SA_Write \| SA_NoAccess\)\]
+$SEC105/558.PostDeref1=\[Post\( Deref=1 \)\]
+$SEC105/559.PostDeref2=\[Post\( Deref=2 \)\]
+$SEC105/560.PostDeref0AccessWrite=\[Post\( Deref=0, Access=Write \)\]
+$SEC105/561.PostDeref1AccessWrite=\[Post\( Deref=1, Access=Write \)\]
+$SEC105/562.PostDeref2AccessRead=\[Post\( Deref=2, Access=Read \)\]
+$SEC105/563.PostDeref2AccessWrite=\[Post\( Deref=2, Access=Write \)\]
+$SEC105/564.PostDeref3AccessWrite=\[Post\( Deref=3, Access=Write \)\]
+$SEC105/565.PostDeref1ValidYes=\[Post\( Deref=1, Valid=Yes \)\]
+$SEC105/566.PostNullYes=\[Post\(Null=Yes\)\]
+$SEC105/567.PostNullYesNullNo=\[Post\(Null=Yes, Null=No\)\]
+$SEC105/568.PostNullSAMaybe=\[Post\(Null=SA_Maybe\)\]
+$SEC105/569.PostNullSAMaybeDeref1=\[Post\(Null=SA_Maybe, Deref=1\)\]
+$SEC105/570.PostValidNo=\[Post\( Valid=No \)\]
+$SEC105/571.PostValidSANo=\[Post\(Valid=SA_No\)\]
+$SEC105/572.PostWritableBytes1=\[Post\(WritableBytes="%1"\)\]
+$SEC105/573.PostWritableBytesConst1=\[Post\(WritableBytesConst=%1\)\]
+$SEC105/574.PostWritableBytesConst1Deref1=\[Post\(WritableBytesConst=%1, Deref=1\)\]
+$SEC105/575.PostDeref1WritableBytesConst1=\[Post\(Deref=1, WritableBytesConst=%1\)\]
+$SEC105/576.PostWritableBytes1Deref1=\[Post\(WritableBytes="%1", Deref=1\)\]
+$SEC105/577.PostValidBytesConst1Deref2=\[Post\(ValidBytesConst=%1, Deref=2\)\]
+$SEC105/578.PostWritableElementsConst1=\[Post\(WritableElementsConst=%1\)\]
+$SEC105/579.PostWritableElementsConst1Deref1=\[Post\(WritableElementsConst=%1, Deref=1\)\]
+$SEC105/580.PostDeref1WritableElementsConst1=\[Post\(Deref=1, WritableElementsConst=%1\)\]
+$SEC105/581.PostWritableElements1Deref1=\[Post\(WritableElements="%1", Deref=1\)\]
+$SEC105/582.PostWritableElements1=\[Post\(WritableElements="%1"\)\]
+$SEC105/583.PostDeref1ValidElementsConst1=\[Post\(Deref=1, ValidElementsConst=%1\)\]
+$SEC105/584.PostDeref1Tainted1=\[Post\(Deref=1, Tainted=%1\)\]
+$SEC105/585.PostValidSAMaybeDeref1=\[Post\(Valid=SA_Maybe, Deref=1\)\]
+$SEC105/586.PostAccess1=\[Post\( Access=%1 \)\]
+$SEC105/587.PostDeref1Access1=\[Post\( Deref=1, Access=%1 \)\]
+$SEC105/588.PostDeref1NeedsRelease1=\[Post\( Deref=1, NeedsRelease=%1 \)\]
```

Comment:
>$SEC105/588 [](http://example.com/codeflow?start=0&length=11)

I don't think this is right. Now every needs to manage hundreds of rules. The help topic for most of these is the same! e.g., 'This annotation is obsolete in v1'. :)

See my other comment. Also, where is the code that is autogenerating all this content, we should check it in.

Reply:
just did as a private method in the Cli project. just call it pointing to the sal.txt and will make the conversion.

It generates a json and a stringbuilder.